vector/src/main/res/layout/activity_media_previewer.xml

```diff
@@ -1,80 +1,80 @@
+        android:id="@+id/toolbar"
+        style="@style/VectorToolbarStyle"
         android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:layout_centerInParent="true"
-        android:scaleType="fitCenter"
-        android:visibility="visible"/>
+        android:layout_height="?attr/actionBarSize"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
 
-    <VideoView
-        android:id="@+id/video_previewer"
+    <FrameLayout
+        android:id="@+id/media_previewer_container"
         android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:layout_centerInParent="true"
-        android:scaleType="fitCenter"
-        android:visibility="visible"/>
+        android:layout_height="0dp"
+        app:layout_constraintBottom_toTopOf="@+id/media_previewer_list"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/toolbar">
 
-    <WebView
-        android:id="@+id/web_previewer"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:layout_centerInParent="true"
-        android:visibility="visible"/>
+        <ImageView
+            android:id="@+id/media_previewer_image_view"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent" />
 
-    <ImageView
-        android:id="@+id/file_previewer"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_centerInParent="true"
-        android:scaleType="fitCenter"
-        android:visibility="visible"
-        android:src="@drawable/filetype_attachment"/>
+        <VideoView
+            android:id="@+id/media_previewer_video_view"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:layout_gravity="center" />
+    </FrameLayout>
 
     <android.support.v7.widget.RecyclerView
-        android:id="@+id/images_preview"
+        android:id="@+id/media_previewer_list"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:layout_alignParentBottom="true"
+        android:layout_marginBottom="8dp"
+        android:layout_marginEnd="8dp"
+        android:layout_marginLeft="8dp"
+        android:layout_marginRight="8dp"
+        android:layout_marginStart="8dp"
         android:background="@color/transparent_dark"
-        android:padding="5dip"/>
+        android:padding="8dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent" />
 
     <TextView
-        android:id="@+id/file_name"
+        android:id="@+id/media_previewer_file_name"
```

Comment:
No need for a background here? (I've not run it)

Reply:
the text is hardly visible, maybe add transparent background

Reply:
Yep, added.